polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -112,9 +112,31 @@ pub(super) fn ends_with(s: &Series, sub: &str) -> PolarsResult<Series> {
     let ca = s.utf8()?;
     Ok(ca.ends_with(sub).into_series())
 }
-pub(super) fn starts_with(s: &Series, sub: &str) -> PolarsResult<Series> {
-    let ca = s.utf8()?;
-    Ok(ca.starts_with(sub).into_series())
+pub(super) fn starts_with(s: &[Series]) -> PolarsResult<Series> {
+    let ca = &s[0].utf8()?;
+    let sub = &s[1].utf8()?;
+
+    match sub.len() {
+        1 => {
+            let out: BooleanChunked = match sub.get(0) {
+                Some(s) => ca.starts_with(s),
+                None => BooleanChunked::new(ca.name(), vec![None; ca.len()]),
+            };
+            Ok(out.into_series())
+        }
+        _ => {
+            let out: BooleanChunked = ca
+                .into_iter()
+                .zip(sub.into_iter())
+                .map(|(opt_src, opt_val)| match (opt_src, opt_val) {
+                    (Some(src), Some(val)) => Some(src.starts_with(val)),
+                    _ => None,
+                })
+                .collect();
```

Comment:
We still have to rename the out array.

So `let mut out ...` and then:

```rust
out.rename(ca.name());
```

Finally, I think we can use `collect_trusted` instead of `collect`, this will be slightly faster.